snippets/coffee-ember-object

```diff
@@ -0,0 +1,243 @@
+
+'.source.coffee':
+
+  'Ember Route':
+    'prefix': 'em.rou'
+    'body':
+      """
+        _export = Ember.Route.extend(
+          model: ->
+            ${1:#@store.find("name_of_a_model")}
+        )
+
+      """
+
+
+
+  'Ember Object Controller':
+    'prefix': 'em.obj'
+    'body':
+      """
+        _export = Ember.ObjectController.extend(
+          ${1:#body}
+        )
+
+      """
+
+  'Ember Array Controller':
+    'prefix': 'em.arr'
+    'body':
+      """
+        _export = Ember.ArrayController.extend(
+          ${1:itemController: 'name_of_an_objectController'}
+          ${2:sortProperties: ['','']}
+          ${3:sortAscending: false}
+          ${4:actions:
+            ${5:name}: ->
+              ${6:#logic...}
+            ${7:name2}: ->
+              ${8:#logic2...}
+            }
+        )
+      """
+
+
+
+
+
+
+  'Ember View':
+    'prefix': 'em.vie'
+    'body':
+      """
+        _export = Ember.View.extend(
+          tagName: ${1:"div"}
+          ${2:classNames: ['','']}
+          ${3:classNameBindings: ['','']}
+          ${4:eventManager: Ember.Object.create(
+            mouseEnter: (event, view) ->
+              #logic...
+              return
+            click:      (event, view) ->
+              #logic...
+              return
+            }
+        )
+
+      """
+
+
+
+
+
+
+  'Ember Component':
+    'prefix': 'em.com'
+    'body':
+      """
+        _export = Ember.Component.extend(
+          ${1:#body}
+        )
+
+      """
+
+
+
+
+
+
+  'Ember Mixin':
+    'prefix': 'em.mix'
+    'body':
+      """
+        _export = Ember.Mixin.create(
+          ${1:#body}
+        )
+
+      """
+
+
+
+
+
+
+  'Ember Object':
+    'prefix': 'em.obj'
+    'body':
+      """
+        _export = Ember.Object.extend(
+          ${1:#body}
+        )
+
+      """
+
+
+
+
+
+
+  'Ember Helper':
+    'prefix': 'em.hel'
+    'body':
+      """
+        _export = Ember.Handlebars.makeBoundHelper( ->
+          ${1:#body}
+        )
+
+      """
+
+
+
+
+
+
+  'Ember Model':
+    'prefix': 'ds.mod'
+    'body':
+      """
+        _export = DS.Model.extend(
+          ${1:title}: DS.attr(${2:"string"})
+          ${3:title: DS.attr("boolean")}
+          ${4:title: DS.attr("number")}
+          ${5:title: DS.belongsTo('someModel',async: true)}
+          ${6:title: DS.hasMany('someModel'  ,async: true)}
```

Comment:
I don't see need for adding all this as part of model. I believe it's better to have it's separate autocomplete.